app/utilities/drawing/master.py

```diff
@@ -0,0 +1,161 @@
+    ))
+
+    # making background rounded rectangle (according to masks/background.png)
+    background = applyAlphaWithMask(background, 'masks/background.png')
+
+    # adding blurry region in the middle of background
+    cropped_img = background.crop((
+        constants.blurry_background_offset[0],
+        constants.blurry_background_offset[1],
+        constants.blurry_region_size[0]+constants.blurry_background_offset[0],
+        constants.blurry_region_size[1]+constants.blurry_background_offset[1]
+    ))
+
+    blurred_img = cropped_img.filter(ImageFilter.GaussianBlur(25)).convert("RGBA")
+    alpha_bg = PIL.Image.new(mode="RGBA", size=blurred_img.size, color=(18, 17, 21, 127))
+    blurred_img = Image.alpha_composite(blurred_img, alpha_bg)
+
+
+    background.paste(
+        blurred_img,
+        (constants.blurry_background_offset[0], constants.blurry_background_offset[1]),
+        create_rounded_rectangle_mask(blurred_img, constants.blurry_region_radius)
+    )
+
+    # adding semi-black-transparent rounded rectangle areas on top of blurried area
+    background = createSemiTransparentBlackRegionOnBackground(
+        constants.black_region_1_size,
+        constants.black_region_1_offset,
+        background
+    )
+    background = createSemiTransparentBlackRegionOnBackground(
+        constants.black_region_2_size,
+        constants.black_region_2_offset,
+        background
+    )
+
+    for i in range(0,7):
+        additional_x_offset = i * 10 + i * constants.badge_size[0]
+        badge_offset = (constants.first_badge_offset[0] + additional_x_offset, constants.first_badge_offset[1])
+        if len(badges_list) > i:
+            background = drawBadgeImageOnBackground(badge_offset, background, badges_list[i])
+        else:
+            background = drawBadgeBackgroundOnBackground(badge_offset, background)
+
+    # creating pfp circle https://stackoverflow.com/a/22336005/11273040 and pasting it on background
+    bigsize = (pfp_original.size[0] * 2, pfp_original.size[1] * 2)
+    mask = Image.new('L', bigsize, 0)
+    draw = ImageDraw.Draw(mask)
+    draw.ellipse((0, 0) + bigsize, fill=255)
+
+    mask = mask.resize(constants.pfp_circle_size, Image.Resampling.LANCZOS)
+    pfp_circle = ImageOps.fit(pfp_original, mask.size, centering=(0.5, 0.5))
+    pfp_circle.putalpha(mask)
+
+    alpha_bg = PIL.Image.new(mode="RGBA", size=background.size, color=(0, 0, 0, 0))
+    alpha_bg.paste(pfp_circle, constants.pfp_circle_offset)
+    background = Image.alpha_composite(background, alpha_bg)
+
+    # progress bar
+    progress_bar_xp = createProgressBar(87, constants.xp_bar_size, background.size, constants.xp_bar_offset)
+    background = Image.alpha_composite(background, progress_bar_xp)
+
+    # drawing texts
+    draw = ImageDraw.Draw(background)
+
+    # checking if nickname provided is longer than 9, if so then it needs to be cut and 3 dots need to be added
+    if len(nickname) > 12:
+        nickname = nickname[0:12] + "..."
```

Comment:
Your comment disagrees with what the code is doing. This is what I think this is meant to do, based on your comment.

```suggestion
    # checking if nickname provided is longer than 12, if so then it needs to be cut by 3 characters and 3 dots need to be added
    if len(nickname) > 12:
        nickname = nickname[0:9] + "..."
```

Reply:
I can see this being useful in multiple places, so I suggest creating a new helper file `/app/utilities/text.py` with this logic as a method `trim_with_ellipsis(text="Foo", max_length=12)`.